quartz_solar_forecast/inverters/enphase.py

```diff
@@ -1,47 +1,154 @@
-    }
-    data = {
-        "grant_type": "client_credentials",
-        "scope": "read"
+        "Authorization": f"Basic {encoded_credentials_str}"
     }
+    conn.request(
+        "POST",
+        f"/oauth/token?grant_type=authorization_code&redirect_uri=https://api.enphaseenergy.com/oauth/redirect_uri&code={auth_code}",
+        payload,
+        headers,
+    )
+    res = conn.getresponse()
+    data = res.read()
+
+    # Decode the data read from the response
+    decoded_data = data.decode("utf-8")
+    # print("UTF-8 DECODED DATA:\n", data.decode("utf-8"))
+
+    # Convert the decoded data into JSON format
+    data_json = json.loads(decoded_data)
+    access_token = data_json["access_token"]
+    # print(access_token)
+    # print(f"The type of access_token is: {type(access_token)}")
+
+    return access_token
 
-    response = requests.post(url, headers=headers, data=data)
-    response.raise_for_status()
-    return response.json()["access_token"]
 
 def get_enphase_data(enphase_system_id: str) -> float:
-    """
+    """ 
     Get live PV generation data from Enphase API v4
-
     :param enphase_system_id: System ID for Enphase API
     :return: Live PV generation in Watt-hours, assumes to be a floating-point number
     """
-    url = f'https://api.enphaseenergy.com/api/v4/{enphase_system_id}/summary'
+    api_key = os.getenv('ENPHASE_API_KEY')
+    access_token = get_enphase_access_token()
+    # print("ACCESS TOKEN: ", access_token)
+    # print("API KEY: ", api_key)
+
+    # Set the start time to 30mins from now
+    start_at = int((datetime.now() - timedelta(minutes=30)).timestamp())
+
+    # start_date = "2024-05-24"
+    # end_date = "2024-05-25"
+
+    # Set the granularity to day
+    granularity = "day"
+
+    conn = http.client.HTTPSConnection("api.enphaseenergy.com")
     headers = {
-        'Authorization': f'Bearer {get_enphase_access_token()}',
-        'key': ENPHASE_API_KEY
+        "Authorization": f"Bearer {str(access_token)}",
+        "key": str(api_key)
     }
 
-    response = requests.get(url, headers=headers)
-    data = response.json()
+    # Add the system_id and duration parameters to the URL
+    url = f"/api/v4/systems/{enphase_system_id}/telemetry/production_micro?start_at={start_at}&granularity={granularity}"
+    conn.request("GET", url, headers=headers)
+
+    # # Printing the GET request
+    # print(f"Request: GET {url}")
+    # print("Headers:")
+    # for header, value in headers.items():
+    #     print(f"{header}: {value}")
+
+    res = conn.getresponse()
+    data = res.read()
+
+    # Decode the data read from the response
+    decoded_data = data.decode("utf-8")
+
+    # Convert the decoded data into JSON format
+    data_json = json.loads(decoded_data)
+    print("DATA: ", data_json)
 
     # Extracting live generation data assuming it's in Watt-hours
-    live_generation_wh = data['current_power']['power']
-    
+    live_generation_wh = data_json['current_power']['power']
```

Comment:
This bit still won't work currently is that right? If so, is it worth adding a TODO comment just above this line that this bit needs be done/fixed still to read in the correct data, just incase someone tries this they know it won't work currently?